### DISCUSSION TITLE
sBTC Core Library / SDK

### DISCUSSION THREAD

**Discussion by stjepangolemac** [stacks-network/stacks-core#3796] [sBTC]
Issues that the proposal is trying to address:

1. sBTC code can be found in multiple repositories
2. sBTC applications and tools need to reimplement core logic of constructing and parsing transactions
3. Work on potential bugs and improvements is duplicated and often involves very disjointed teams of people

The core-eng team at Trust Machines has talked about this a few times and the proposal is simple:

1. Create a core sBTC library that contains everything
  1.1 Wire formats
  1.2 Pick between OP_RETURN and commit-reveal modes
  1.3 Transaction construction
  1.4 Transaction parsing
  1.5 Testing
2. CLI interface
  2.1 Build deposit and withdrawal transactions
  2.2 Broadcast transactions
  2.3 Create credentials
3. Optional - expose the core through different mechanisms to other languages
  3.1 WASM - for usage in NodeJS, browser, and other compatible languages
  3.2 pyo3 - for usage in Python
  3.3 FFI, CXX, etc
4. Use the core library in stacks-node, bridge, and all other places that relate to sBTC

We'd like some discussion to happen before we're ready to commit to this work. Do you think this is a sensible idea that would produce value for the ecosystem?

**Comment by netrome** [comment]
I think this will be very helpful for sBTC moving forward. This will allow us to validate and test future sBTC versions (sBTC Mini, final release and anything in between), and would make it much easier for developers to integrate their apps with sBTC.

Thank you for writing this up @stjepangolemac. Let's await more feedback before starting designing and planning for this. There's a lot of folks I can't ping here, so I'll link this discussion in the [Stacks discord](https://discord.com/channels/621759717756370964/1128015593510023320).

**Comment by stjepangolemac** [comment]
We have a repo https://github.com/stacks-network/sbtc 🎉

**Reply by netrome** [reply]
Niiiice! 🚀
